metaflow/plugins/cards/ui/src/utils.ts

```diff
@@ -43,28 +33,60 @@ export const getPageHierarchy = (
 /**
  * Function to calculate pixels from computed rem size
  */
-export const convertPixelsToRem = (px: number): number => {
-  return (
-    px /
-    parseFloat(
-      getComputedStyle(document.documentElement).fontSize.replace("px", "")
-    )
-  );
+export const convertPixelsToRem = (px: number, doc?: Document): number => {
+  // adding default to allow for testing where no document exists
+  const computedStyle = doc
+    ? parseFloat(
+        getComputedStyle(document.documentElement).fontSize.replace("px", "")
+      )
+    : 16;
+
+  return px / computedStyle;
+};
+
+/**
+ * Our pathspecs are made up of flowname/runid/stepname/taskid,
+ * where stepname/taskid are optional.  So to keep this testable and predictable,
+ * we have 2 functions, one to get the object, and another to get a key.  You can
+ * use whichever you need the data for
+ */
+export const getPathSpecObject = (pathspec: string): types.PathSpecObject => {
```

Comment:
Add another type so that we can only pass "flowname" | "runid" etc

Reply:
can you confirm what you mean here?  the path spec should be the full pathspec noted in the comment above it.  

however, if you see below, I am requiring a key types.PathSpecObject, so I do believe we'll get that advantage.

Reply:
yep. That's good. I saw the `string` and freaked